### DISCUSSION TITLE
Invalid next.config.js options detected unexpected property - fallback

### DISCUSSION THREAD

**Discussion by kirillleogky** [vercel/next.js#43635] [Help] [answered]
```
    "next": "^12.3.0",
    "next-auth": "^4.2.1",
    "next-i18next": "^13.0.0",
    "next-progress": "^2.2.0",
    "next-pwa": "^5.6.0",
```

 </br>

 </br>

`next.config.js`

```
/** @type {import('next').NextConfig} */
const { i18n } = require('./next-i18next.config');

const withPWA = require('next-pwa')({
  dest: 'public',
  swSrc: 'service-worker.js',
  disable: process.env.NODE_ENV === 'development',
  // cacheOnFrontEndNav: true,
  fallbacks: {
    image: '/assets/default-image.png',
  },
});

const nextConfig = {
  i18n,
  // Due to Material UI styles mismatch
  reactStrictMode: false,
  fallback: true,                                            <--- this prop
  images: {
    remotePatterns: [
      {
        protocol: 'https',
        hostname: '**',
      },
    ],
  },

  compiler: {
    emotion: true,
  },

  publicRuntimeConfig: {
     ......
  },

  serverRuntimeConfig: {
    .....
  },

  webpack(config) {
    config.module.rules.push({
      test: /\.svg$/,
      use: ['@svgr/webpack'],
    });

    return config;
  },

  async redirects() {
    return [
      {
        source: '/admin',
        destination: '/admin/performance',
        permanent: true,
      },
      {
        source: '/creator',
        destination: '/creator/episodes',
        permanent: true,
      },
    ];
  },
};

module.exports = withPWA(nextConfig);

```
 </br>
 
Where I need to place `fallback: true` ?
 </br>
With `"next": "^12.1.5"` I didn't have error

![image](https://user-images.githubusercontent.com/47118725/205264367-91c8fc66-9c32-4d65-9533-28fa257b5e73.png)

**Comment by icyJoseph** [comment, marked as answer]
Starting with 12.1.6 the next.config.js loading step also checks for invalid properties. What was `fallback` supposed to do? 

Looking at https://github.com/vercel/next.js/blob/canary/packages/next/server/config-shared.ts#L158 there's no fallback option what so ever, but maybe there was in order versions?

**Reply by kirillleogky** [reply]
> Starting with 12.1.6 the next.config.js loading step also checks for invalid properties. What was `fallback` supposed to do?
> 
> Looking at https://github.com/vercel/next.js/blob/canary/packages/next/server/config-shared.ts#L158 there's no fallback option what so ever, but maybe there was in order versions?

I didn't write this but I thought that it is required to display page/fallback (if we don't have access to the internet)

Mine `pages/fallback.tsx`:

```
// libraries
import React from 'react';
import { serverSideTranslations } from 'next-i18next/serverSideTranslations';
import { GetStaticProps } from 'next/types';
import { Typography, Box } from '@mui/material';
import { useTranslation } from 'next-i18next';

const Fallback = () => {
  const { t } = useTranslation();

  return (
    <Box
      sx={{
        textAlign: 'center',
      }}
    >
      <Typography variant="h1">{t('errors:offline')}</Typography>
      <Typography variant="h3">{t('errors:connectionError')}</Typography>
    </Box>
  );
};

export const getStaticProps: GetStaticProps = async ctx => {
  const { locale } = ctx;

  return {
    props: {
      ...(await serverSideTranslations(locale as string, ['navbar', 'custom-popover', 'notification', 'errors'])),
    },
  };
};

export default Fallback;

```

**Reply by icyJoseph** [reply]
As far as I know, there's no such thing at that level. This just means that `/fallback` is a page we can visit. 

From what I can see, it is not needed, as next.config.js says itself, it is invalid. But this idea that `pages/fallback` would be used when offline, and when next.config.js has fallback true, is something I never heard of.

**Reply by kirillleogky** [reply]
> As far as I know, there's no such thing at that level. This just means that `/fallback` is a page we can visit.
> 
> From what I can see, it is not needed, as next.config.js says itself, it is invalid. But this idea that `pages/fallback` would be used when offline, and when next.config.js has fallback true, is something I never heard of.

thank you Joseph)
I'll try to remove it
